crates/biome_js_analyze/src/lint/style/no_restricted_imports.rs

```diff
@@ -315,6 +332,93 @@ declare_lint_rule! {
     /// ```js,use_options
     /// import { Bar } from 'import-bar';
     /// ```
+    ///
+    /// ### `patterns`
+    ///
+    /// This option allows you to specify multiple modules to restrict using gitignore-style patterns.
+    ///
+    /// ### `group`
+    ///
+    /// The patterns array can also include objects. The group property is used to specify the gitignore-style patterns for restricting modules and the message property is used to specify a custom message.
+    ///
+    /// ```json,options
+    /// {
+    ///     "options": {
+    ///         "patterns": [{
+    ///             "group": ["import-foo/*", "!import-foo/bar"],
+    ///             "message": "import-foo is deprecated, except the modules in import-foo/bar."
+    ///         }]
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// #### Invalid
+    ///
+    /// ```js,expect_diagnostic,use_options
+    /// import foo from 'import-foo/foo';
+    /// ```
+    ///
+    /// #### Valid
+    ///
+    /// ```js,use_options
+    /// import foo from 'import-foo';
+    /// import bar from 'import-foo/bar';
+    /// ```
+    ///
+    /// ### `importNamePattern`
+    ///
+    /// This option allows you to use regex patterns to restrict import names.
+    ///
```

Comment:
```suggestion
    /// ### `importNamePattern`
    /// **Since `v2.1.0`**
    ///
    /// This option allows you to use regex patterns to restrict import names.
    ///
```